### PR TITLE
Add Checkov action to CI/CD pipeline

### LINKED ISSUE #980: Add Checkov to CI/CD pipeline
# 💡 Summary #

Add Checkov to a workflow and the workflow to the CI/CD pipeline.

## Motivation and context ##

Checkov is a venerable static code analysis tool that is useful for finding all sorts of security issues in a repo.

This would be useful because security is good.

## Implementation notes ##

Please provide details for implementation, such as:

- Use ML, add to workflow.
- Add workflow to pipeline.
- Test
- Check current results to determine which are real and which are false positives.
- Fix real issues, comment out false positive.
- Test some more.

## Acceptance criteria ##

How do we know when this work is done?

- [ ] Checkov is checking our repo.
- [ ] All security issues (that it finds) have been resolved.

### PR DESCRIPTION
**Pull request by james-garriss**
## 🗣 Description ##

The goal of this PR was to add the Checkov action to the CI/CD pipeline.  While there, I also slightly reorg'd the pipeline to fail fast and rewrote the PS linter action (reduced the runtime from 3m30s to 1m).

Closes #980

## 💭 Motivation and context ##

This is part of our efforts to build a robust CI/CD pipeline that builds, tests, and releases ScubaGear.

## 🧪 Testing ##

I ran the pipeline, fixed a security bugs, and saw that everything was a lovely shade of green.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [x] Changes are limited to a single goal - *eschew scope creep!*

Well, I did have a little bit of scope creep, as improving the PS linter wasn't technically part of the task.  But I really wanted the pipeline to run faster.  Now it does.  :-D

- [x] Changes are sized such that they do not touch excessive number of files.
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [x] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All relevant repo and/or project documentation updated to reflect these changes.
- [x] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [x] PR passed smoke test check.
- [x] Feature branch has been rebased against changes from parent branch, as needed

  Use `Rebase branch` button below or use [this](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) reference to rebase from the command line.
- [x] Resolved all merge conflicts on branch
- [x] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

File: .github/workflows/lint_powershell.yaml [hunk 1]
```diff
@@ -7,21 +7,27 @@ on:
...
       run:
-        shell: bash
+        shell: pwsh
     steps:
       - name: Checkout Repo
         uses: actions/checkout@v4
-      - name: Setup Config File
-        run: cp Testing/Linting/MegaLinter/.mega-linter-powershell.yml .mega-linter.yml
+        # - name: Setup Config File
+        #   run: cp Testing/Linting/MegaLinter/.mega-linter-powershell.yml .mega-linter.yml
```
[the remark this answers is not in the document]
Comment (james-garriss) [reply .github/workflows/lint_powershell.yaml:26]: Removed.

File: .github/workflows/lint_powershell.yaml [hunk 2]
```diff
@@ -7,21 +7,27 @@ on:
...
       - name: Setup PSScriptAnalyzer settings
         run: cp Testing/Linting/MegaLinter/.powershell-psscriptanalyzer.psd1 .powershell-psscriptanalyzer.psd1
-      - name: Lint PowerShell
-        uses: oxsecurity/megalinter/flavors/dotnet@latest
+        # - name: Lint PowerShell
+        #   uses: oxsecurity/megalinter/flavors/dotnet@latest
```
[the remark this answers is not in the document]
Comment (james-garriss) [reply .github/workflows/lint_powershell.yaml:30]: Removed.

File: PowerShell/ScubaGear/Testing/Unit/PowerShell/CreateReport/CreateReportStubs/ProviderSettingsExport.json
[the remark this answers is not in the document]
Comment (james-garriss) [reply PowerShell/ScubaGear/Testing/Unit/PowerShell/CreateReport/CreateReportStubs/ProviderSettingsExport.json:1]: Yes.  I addd this around like 16,000:

"checkov:skip=CKV_SECRET_3:This is not an Azure Storage Account access key.",

**Review by dagarwal-mitre — APPROVED** [review]

**Comment by james-garriss** [comment]
@nanda-katikaneni , this one is ready.  Thanks!

**Review by vinniespina — COMMENTED** [review]
i commit

**Review by vinniespina — COMMENTED** [review]
i commit

**Review by vinniespina — COMMENTED** [review]
i commit

**Pull request merged**